zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/AuthorizationService.java

```diff
@@ -259,28 +261,84 @@ public Set<String> getRoles(String user) {
+    Set<String> writers = new HashSet<>(getWriters(noteId));
+    writers.addAll(getDefaultWriters());
+    return isMember(entities, writers) ||
+           isMember(entities, owners) ||
+           isAdmin(entities);
   }
 
   public boolean isReader(String noteId, Set<String> entities) {
-    return isMember(entities, getReaders(noteId)) ||
-            isMember(entities, getOwners(noteId)) ||
-            isMember(entities, getWriters(noteId)) ||
-            isMember(entities, getRunners(noteId)) ||
-            isAdmin(entities);
+    Set<String> owners = new HashSet<>(getOwners(noteId));
+    owners.addAll(getDefaultOwners());
+    Set<String> writers = new HashSet<>(getWriters(noteId));
+    writers.addAll(getDefaultWriters());
+    Set<String> runners = new HashSet<>(getRunners(noteId));
+    runners.addAll(getDefaultRunners());
+    Set<String> readers = new HashSet<>(getReaders(noteId));
+    readers.addAll(getDefaultReaders());
+    return isMember(entities, readers) ||
+           isMember(entities, owners) ||
+           isMember(entities, writers) ||
+           isMember(entities, runners) ||
+           isAdmin(entities);
   }
 
   public boolean isRunner(String noteId, Set<String> entities) {
-    return isMember(entities, getRunners(noteId)) ||
-            isMember(entities, getWriters(noteId)) ||
-            isMember(entities, getOwners(noteId)) ||
-            isAdmin(entities);
+    Set<String> owners = new HashSet<>(getOwners(noteId));
+    owners.addAll(getDefaultOwners());
+    Set<String> writers = new HashSet<>(getWriters(noteId));
+    writers.addAll(getDefaultWriters());
+    Set<String> runners = new HashSet<>(getRunners(noteId));
+    runners.addAll(getDefaultRunners());
+    return isMember(entities, runners) ||
+           isMember(entities, writers) ||
+           isMember(entities, owners) ||
+           isAdmin(entities);
+  }
+
+  private Set<String> getDefaultOwners() {
+    return getDefaultRoles(ZeppelinConfiguration.ConfVars.ZEPPELIN_OWNER_ROLES);
+  }
+
+  private Set<String> getDefaultWriters() {
+    return getDefaultRoles(ZeppelinConfiguration.ConfVars.ZEPPELIN_WRITER_ROLES);
+  }
+
+  private Set<String> getDefaultReaders() {
+    return getDefaultRoles(ZeppelinConfiguration.ConfVars.ZEPPELIN_READER_ROLES);
+  }
+
+  private Set<String> getDefaultRunners() {
+    return getDefaultRoles(ZeppelinConfiguration.ConfVars.ZEPPELIN_RUNNER_ROLES);
+  }
+
+  private Set<String> getDefaultRoles(ZeppelinConfiguration.ConfVars confvar) {
+    Set<String> defaultRoles = new HashSet<>();
+    String defaultRolesConf = null;
+    switch (confvar) {
+      case ZEPPELIN_OWNER_ROLES:
+      case ZEPPELIN_WRITER_ROLES:
+      case ZEPPELIN_READER_ROLES:
+      case ZEPPELIN_RUNNER_ROLES:
+        defaultRolesConf = zConf.getString(confvar);
+        break;
+      default:
+        LOGGER.warn("getDefaultRoles is used with {}, which is not valid", confvar);
+        break;
+    }
```

Comment:
How about using a collection like EnumSet instead of a switch statement to manage valid roles? IMO, this simplifies the code and makes it easier to add or modify roles in the future.

```java
private static final Set<ZeppelinConfiguration.ConfVars> VALID_CONF_VARS = EnumSet.of(
      ZeppelinConfiguration.ConfVars.ZEPPELIN_OWNER_ROLES,
      ZeppelinConfiguration.ConfVars.ZEPPELIN_WRITER_ROLES,
      ZeppelinConfiguration.ConfVars.ZEPPELIN_READER_ROLES,
      ZeppelinConfiguration.ConfVars.ZEPPELIN_RUNNER_ROLES
  );

private Set<String> getDefaultRoles(ZeppelinConfiguration.ConfVars confvar) {
    if (!VALID_CONF_VARS.contains(confvar)) {
      LOGGER.warn("getDefaultRoles is used with {}, which is not valid", confvar);
      return Collections.emptySet();
    }
    ...
}
```

Reply:
Thank you for your review. I have adapted the code and agree with your statements.